benchexec/tablegenerator/react-table/src/components/ReactTable.js

```diff
@@ -641,12 +641,22 @@ const Table = (props) => {
 
   // get selected status and category values
   useEffect(() => {
+    // To check if any task filter is applied
+    let taskFilterExists = false;
+
     for (const filter of props.filters) {
       const { values, id } = filter;
-      if (id === "id") {
-        setDisableTaskText(!isNil(values));
+      if (id === "id" && !isNil(values)) {
+        taskFilterExists = true;
+        setDisableTaskText(true);
       }
     }
+
+    // Enable task text when task filter is not applied
+    if (!taskFilterExists) {
+      setDisableTaskText(false);
+    }
```

Comment:
Wouldn't it work if we just call `setDisableTaskText` once in the end (after the loop) with the result of whether a task-filter exists? I guess this would be easier to understand when reading the code.

We could also maybe use functional style with `some()` instead of a manual loop?

Reply:
Yes, I have replaced the loop with `some()`.